### DISCUSSION TITLE
Replace text from one file into another in r (possible to python?)

### DISCUSSION THREAD

**Discussion by Saupia** [teamalgoritma/community#490] [Q&A] [answered]
Dear Data Maniac, 

I need to replace a text from X file into a Y file for a Classification study. "Cyberbullying". 
I run the cleaning for slang using `textclean` library but it does not work. At the same time with `katadasar`. 
THe problem is in bold in the code. 
Is it possible to manual cleaning for the slank in Python?
I have 2 different set of data and I need to replace them. 

`---
title: "cyberbulling-twitter"
author: "RAR"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Project Goal
Make aprediction to classify cyberbully text in Indonesian. 

#Library
```{r}
library(dplyr)
library(grid)
library(gtools)
library(e1071)
library(tm)
library(SnowballC)
library(ROCR)
library(partykit)
library(caret)
library(class)
library(gmodels)
library(devtools)
library(ggplot2)
library(katadasaR)
library(textclean)
library(katadasaR)
library(wordcloud)
library(tokenizers)
```

install

#Read data
```{r}
raw_data <- read.csv("train.csv", 
         sep = ",",
         stringsAsFactors = FALSE,
         encoding = "UTF-8") 
head(raw_data)
```
The data has 10,535 rows and 8 columns.

`bully`: Whether the tweet is classified as cyberbully (yes) or not
`tweet`: The content of the tweet.
`individual`: Whether the tweet is a cyberbully targeted toward certain individual (0 = no, 1 = yes)
`group`: Whether the tweet is a cyberbully targeted toward certain group of people (0 = no, 1 = yes)
`gender`: Whether the tweet is a cyberbully based on gender or cursing someone using words that are degrading to gender (0 = no, 1 = yes)
`physical`: Whether the tweet is a cyberbully based on physical deficiencies/differences or disability (0 = no, 1 = yes)
`race`: Whether the tweet is a cyberbully based on a human race or ethnicity (0 = no, 1 = yes)
`religion`: Whether the tweet is a cyberbully based on a religion, religious organization, or a particular creed (0 = no, 1 = yes)

#Data Wrangling

```{r}
cyber <- 
  raw_data %>% 
  mutate(bully = as.factor(bully)) %>% 
  mutate_if(is.integer, as.factor)
```

*Import Indonesian Slang*
```{r}
spell_lex <- read.csv2("lexicon_indonesian.csv", sep = ",")
```

***Replace Interenet Slang***
```{r}
**cyber <- replace_internet_slang(cyber, slang = paste0("\\b", spell_lex$slang, "\\b"), replacement = spell_lex$formal, ignore.case = TRUE)**
```

```{r}
kata <- c("makanan", "minuman", "pegunungan", "menggurui")
# sapply(kata, katadasar)
```

```{r}
# cyber_kd <- lapply(cyber$tweet, katadasaR) #FAILED
```

#Tokenize and Conjunction removal

```{r}
cyber <- tokenize_words (cyber)
```

#Exploratory Data Analysis

```{r}
prop.table(table(cyber$bully))
```

```{r}
summary(cyber)
```
Insights
1. `individual` is a category with the most bullying text. 
2. 

```{r}
cyber %>% filter(bully %in% "yes") %>% 
  sample_n(3)
```

##Data Transformation to Corpus

```{r}
cyber_corpus <- VCorpus(VectorSource(cyber$tweet))
```
##Inspecting data

*observing text through dataframe*
```{r}
cyber[100,2]
```
*Observing text through corpus document*
```{r}
cyber_corpus[[100]]$content
```

```{r}
lapply(cyber_corpus[1:2]$content, as.character)
```
*Case-Folding: Change all of the capitalization into lowercase*
```{r}
cyber_corpus <- tm_map(x = cyber_corpus,
                       FUN = content_transformer(tolower))
cyber_corpus[[218]]$content
```

*Remove all of the numbers and all of the conjunction*

```{r}
cyber_corpus <- tm_map(x = cyber_corpus,
                       FUN = removeNumbers)
cyber_corpus[[218]]$content
```

*Remove punctuation*

```{r}
  transformer <-  content_transformer(FUN = function(x, pattern){gsub(x = x, pattern = pattern, replacement = " " )
    })
```

```{r}
cyber
```

```{r}
cyber_corpus <- tm_map(cyber_corpus, transformer, "^")
cyber_corpus <- tm_map(cyber_corpus, transformer, "@")
cyber_corpus <- tm_map(cyber_corpus, transformer, "_")
cyber_corpus <- tm_map(cyber_corpus, transformer, ".")
cyber_corpus <- tm_map(cyber_corpus, transformer, "!")
cyber_corpus <- tm_map(cyber_corpus, transformer, "?")
cyber_corpus <- tm_map(cyber_corpus, transformer, ",")
cyber_corpus <- tm_map(cyber_corpus, transformer, "#")
cyber_corpus <- tm_map(cyber_corpus, transformer, ";")
cyber_corpus <- tm_map(cyber_corpus, transformer, "�")
cyber_corpus <- tm_map(cyber_corpus, transformer, ")")
cyber_corpus <- tm_map(cyber_corpus, transformer, "/")
cyber_corpus <- tm_map(cyber_corpus, transformer, "]")
cyber_corpus <- tm_map(cyber_corpus, transformer, "%")
cyber_corpus <- tm_map(cyber_corpus, transformer, "*")
cyber_corpus <- tm_map(cyber_corpus, transformer, "`")
cyber_corpus <- tm_map(cyber_corpus, transformer, "-")
cyber_corpus <- tm_map(cyber_corpus, transformer, "-")

```

`

**Comment by kinantytasyao** [comment, marked as answer]
Dear Pak @Saupia 

I've reviewed your code and i found the problem is in the chunk:

`cyber <- replace_internet_slang(cyber, slang = paste0("\\b", spell_lex$slang, "\\b"), replacement = spell_lex$formal, ignore.case = TRUE)`

`cyber <- tokenize_words(cyber)`

in the code above you need to replace all `cyber` (dataframe) with a vector containing the tweets you want to process (`cyber$tweet`) 

hope this answer will help :)

**Reply by Saupia** [reply]
Semoga bisa. 
Terima kasih.
